src/mem.c

```diff
@@ -32,11 +32,13 @@ coap_memory_init(void) {
 
 void *
 coap_malloc_type(coap_memory_tag_t type UNUSED_PARAM, size_t size) {
+  (void)type;
```

Comment:
If you'd like, I could define something like a *COAP_UNUSED*() macro, to make the intentions more clear.